com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -1013,9 +1013,22 @@ else if (build_report_html != null) {
     }
 
     private static void logErrorAndExit(Exception e) {
-        System.err.println(e.getMessage());
-        if (e.getCause() != null) {
-            System.err.println("Cause: " + e.getCause());
+        logger.severe(e.getMessage().toString());
+        Throwable cause = e.getCause();
+        if(cause != null) {
+            for(int i = 0; cause != null; ++i) {
+                logger.severe("Cause:%d: %s", i, cause.toString());
+                StackTraceElement[] stackTrace = cause.getStackTrace();
+                for (StackTraceElement element : stackTrace) {
+                    logger.severe(element.toString());
+                }
+                cause = cause.getCause();
+            }
+        } else {
+            StackTraceElement[] stackTrace = e.getStackTrace();
+            for (StackTraceElement element : stackTrace) {
+                logger.severe(element.toString());
+            }
```

Comment:
Please add short comments to your pr that mentions the problem you're trying to solve, as it's hard to at a glance get into the correct context of the change.

Also, we like to keep separate changes apart, for easier maintenance.
So, in future, this would be in a separate PR.

Reply:
I split it into a separate commit. The real problem here was that throw would happen and then get rethrown but when it was finally caught in the default handler it was very hard to see where the original came from. So I just added logging for cause to make it easier to get back to code that caused the initial exception.